src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -402,14 +399,20 @@ private static DeclarationModifiers ToDeclarationModifier(SyntaxKind kind)
                     ref seenNoDuplicates,
                     diagnostics);
 
-                if (one == DeclarationModifiers.Partial && i < modifiers.Count - 1)
+                if (one == DeclarationModifiers.Partial)
                 {
-                    // There was a bug where we allowed `partial async` at the end of modifiers on methods. We keep this behavior for backcompat.
-                    if (!(isOrdinaryMethod && i == modifiers.Count - 2 && ToDeclarationModifier(modifiers[i + 1].ContextualKind()) == DeclarationModifiers.Async))
+                    var messageId = isForTypeDeclaration ? MessageID.IDS_FeaturePartialTypes : MessageID.IDS_FeaturePartialMethod;
+                    messageId.CheckFeatureAvailability(diagnostics, modifier.Parent, modifier.GetLocation());
+
+                    if (i < modifiers.Count - 1)
```

Comment:
This test seems to be subsumed by the later test `i == modifiers.Count - 2`.

Reply:
indeed.  i was preserving it since it was in the original code, but it looks like it can eb removed.  will remove in followup.

Reply:
nope.  it's necessary.  the innert check is negated.  i tried removing this and broke *tons* :)

Reply:
i've tweaked the code to make the logic clearer.